internal/services/hybridcompute/hybrid_compute_machine_data_source.go

```diff
@@ -533,6 +465,9 @@ func (r HybridComputeMachineDataSource) Read() sdk.ResourceFunc {
 	return sdk.ResourceFunc{
 		Timeout: 5 * time.Minute,
 		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			logger := sdk.ConsoleLogger{}
+			logger.Warn("Datasource azurerm_hybrid_compute_machine will be deprecated and replaced by azurerm_arc_machine in 4.0 and later version.")
```

Comment:
This isn't how we do deprecations, and also this would be output as a debug warning only, instead we'll need to add [an interface similar to this for Resources](https://github.com/hashicorp/terraform-provider-azurerm/blob/b3c8d0f060687192d072c89fdf4c2e08b6cb544c/internal/sdk/resource.go#L109-L120) (but for Data Sources) and then [conditionally expose that here](https://github.com/hashicorp/terraform-provider-azurerm/blob/b3c8d0f060687192d072c89fdf4c2e08b6cb544c/internal/sdk/wrapper_data_source.go#L53-L55), in the same way [that we do for Resources](https://github.com/hashicorp/terraform-provider-azurerm/blob/b3c8d0f060687192d072c89fdf4c2e08b6cb544c/internal/sdk/wrapper_resource.go#L155-L166))

Can we update this?

Reply:
Thanks for the advice, I will update the interface.